inference-engine/src/gna_plugin/gna_plugin_internal.hpp

```diff
@@ -32,6 +33,7 @@ class GNAPluginInternal  : public InferenceEngine::InferencePluginInternal {
     InferenceEngine::ExecutableNetworkInternal::Ptr LoadExeNetworkImpl(
                                                 const InferenceEngine::CNNNetwork &network,
                                                 const std::map<std::string, std::string> &config) override {
+        std::lock_guard<std::mutex> lock{ syncCallsToLoadExeNetworkImpl };
```

Comment:
why do we need it? It's better to remove / minimize global state and allow to call `LoadNetwork` in parallel..

E.g. I suppose it least GNA passes can be run in parallel, while an access to device can be synchronized..

Reply:
@ilya-lavrenov, the scenario when several LoadNetwork calls must be run for GNA plugin in parallel with maximum performance seems absolutely unrealistic. I would suggest we merge this solution.